misc: Update README/README.md

BobbyRBruce opened the pull request:
Change-Id: Ic9174d6c3d905f85e2f7f159478d8c93ef7e4079

BobbyRBruce committed:
misc: Update README to README.md

This change converts the text-based README to markdown. This works
better with modern source-control systems, most notably, GitHub.

The README.md has been broken down into sections to better organize the
document.

Change-Id: I18018ba23493f43861544497f23ec59f1e8debe1

BobbyRBruce committed:
misc: Update README.md "gem5 Resources" section

Change-Id: I90296c69098e9bf5987c142d52ab57d556a22009

BobbyRBruce committed:
misc: Update "Getting Help" section

This section now included expanded information on Reporting bugs and
Requesting Features.

Change-Id: I555f9f5edb683a1ee235058045fbce07ea473138

BobbyRBruce committed:
misc: Add blurb to README.md intro

Change-Id: Ic9174d6c3d905f85e2f7f159478d8c93ef7e4079

BobbyRBruce committed:
scons: Update infopy.py to handle '.' in filename

Due to renaming 'README' to 'README.md', this code was generating the
following for "info.py":

```
README.md = "<FILE CONTENTS HERE>"
```

As '.' is used to access member variables/methods in python. To fix this
"infopy.oy" now replaces "." with "_". As such the generated in in
"info.py" is now:

```
README_MD = "<FILE CONTENTS HERE>"

Change-Id: I834fa84f654b27ef399a31cda545d2bf2d906390

powerjg commented on README.md:
```diff
@@ -0,0 +1,88 @@
+
+Once you have all dependencies resolved, type
+`scons build/<CONFIG>/gem5.opt` where `<CONFIG>` is one of the options in
+build_opts like `ARM`, `NULL`, `MIPS`, `POWER`, `SPARC`, `X86`,
```
ALL should be the default here now.

powerjg commented on README.md:
```diff
@@ -0,0 +1,88 @@
+* ext: less-common external packages needed to build gem5
+* include: include files for use in other programs
+* site_scons: modular components of the build system
+* src: source code of the gem5 simulator
```
```suggestion
* src: source code of the gem5 simulator. The C++ source, Python wrappers, and Python standard library are found in this directory.
```

powerjg commented on README.md:
```diff
@@ -0,0 +1,88 @@
+* **gem5-dev@gem5.org**: A mailing list for developers of gem5 to ask questions
+or start discussions. To join the mailing list please visit
+<https://www.gem5.org/mailing_lists>.
+* **GitHub Discussions**: A GitHub Discussions page. This can be used to start
```
Reorder with this at the top? Do we want to encourage github first and mailing list last?

Similarly, maybe we should move slack up?

BobbyRBruce replied:
I've reordered this.

powerjg commented on build_tools/infopy.py:
```diff
@@ -51,7 +51,7 @@
 
 for source in args.files:
-    src = os.path.basename(source)
+    src = os.path.basename(source).replace(".", "_")
```
A comment for why this is needed would be good.

powerjg commented on src/SConscript:
```diff
@@ -615,7 +615,7 @@ PySource('m5', 'python/m5/defines.py')
 # Generate a file that wraps the basic top level files
 gem5py_env.Command('python/m5/info.py',
-            [ File('#/COPYING'), File('#/LICENSE'), File('#/README'),
+            [ File('#/COPYING'), File('#/LICENSE'), File('#/README.md'),
```
This is OK for now. However, I think we should just remove `info.py` completely. I can't think of any other tool that has the readme/license embedded this way.

BobbyRBruce replied:
> This is OK for now. However, I think we should just remove `info.py` completely. I can't think of any other tool that has the readme/license embedded this way.

I'm not sure I know exactly why we do this at all. I assume we print this to a terminal somewhere? Any way, seems weird. I'll put it in an issue.

BobbyRBruce replied:
Issue #89 is to remove this at some point.

BobbyRBruce committed:
misc: Expand 'src' desc in README.md

Co-authored-by: Jason Lowe-Power <jason@lowepower.com>

BobbyRBruce committed:
misc: Update 'Building gem5' to default to `ALL` comp

Change-Id: I23f204507be0908afa982712adc9ff7b6c7e8dd3
Addresss: https://github.com/gem5/gem5/pull/71#discussion_r1263932939

BobbyRBruce committed:
misc: Reorder "Getting Help" list

This puts GitHub Discussions and GitHub Issues towards the top of the
list. This is to incentivize their usage.

Change-Id: I248e3a5ace58bf909182300a3c44a412a8d73e6e
Addresses: https://github.com/gem5/gem5/pull/71#discussion_r1263934519

BobbyRBruce committed:
scons: Add comment explaining "." -> "_" replace

Change-Id: Ibfb4642768cc29a5da04ae416ff137ad8bf11b68
Addresses: https://github.com/gem5/gem5/pull/71#discussion_r1263934875

BobbyRBruce commented:
BTW: I didn't mean to delete that comment. It was you asking to make `ALL/gem5.opt` the default. I did so.
